control/navigation.c

```diff
@@ -636,6 +643,30 @@ static void navigation_critical_handler(navigation_t* navigation)
 	}
 }
 
+static void navigation_throw_recovery_position_hold_handler(navigation_t* navigation)
+{
+	float rel_pos[3];
+
+	if (!navigation->throw_recovery_position_set)
+	{
+		navigation->throw_recovery_position_set = true;
+
+		print_util_dbg_print("Cust: height control to 2m");
+		navigation->state->mav_mode_custom &= 0xFFFFFFE0;
+		navigation->state->mav_mode_custom = CUST_THROW_RECOVERY_POSITION_HOLD;
+		navigation->waypoint_handler->waypoint_hold_coordinates = navigation->position_estimation->local_position;
+		navigation->waypoint_handler->waypoint_hold_coordinates.pos[Z] = -2.0f;
+
+		for (uint8_t i = 0; i < 3; i++)
+		{
+			rel_pos[i] = navigation->waypoint_handler->waypoint_critical_coordinates.pos[i] - navigation->position_estimation->local_position.pos[i];
```

Comment:
typo here I guess.
you are using waypoint_hold_coordinates and not critical one